src/app/qgslayerstylingwidget.cpp

```diff
@@ -444,6 +438,10 @@ void QgsLayerStylingWidget::updateCurrentWidgetLayer()
     {
       mMesh3DWidget = widget;
     }
+    else if ( QgsPointCloudLayer3DRendererWidget *widget = qobject_cast<QgsPointCloudLayer3DRendererWidget *>( current ) )
```

Comment:
You can remove all the changes to this file -- they are handled automatically through the factory api

Reply:
I missed that I need to override supportsStyleDock in QgsPointCloudLayer3DRendererWidgetFactory for the factory to work.
I think it is good now